geodiff/src/geodiff.cpp

```diff
@@ -1210,7 +1210,13 @@ bool GEODIFF_CT_columnIsPkey( GEODIFF_ContextH /*contextHandle*/, GEODIFF_Change
 
 const char *GEODIFF_createWkbFromGpkgHeader( GEODIFF_ContextH contextHandle, const char *gpkgWkb, size_t *length )
 {
-  auto result = createWkbFromGpkgHeader( std::string( gpkgWkb, *length ) );
+  const Context *context = static_cast<const Context *>( contextHandle );
+  if ( !context )
```

Comment:
as I write, I think the function should return int (geodiff success or error) and the result should go as 3rd argument before lenght